include/deal.II/grid/grid_generator.h

```diff
@@ -358,6 +358,19 @@ namespace GridGenerator
                    const double        radius = 1.);
 
   /**
+    * Creates a codimension one hyper ball.
+    * This function is declared to exist for dim=1, spacedim =2, and
+    * dim=2, spacedim =3.
+    %
+    * @note The triangulation needs to be void upon calling this function.
+    */
+
+  template <int dim, int spacedim>
+  void hyper_ball (Triangulation<dim,spacedim> &tria,
+                   const Point<spacedim>   &center = Point<spacedim>(),
+                   const double        radius = 1.);
```

Comment:
Why don't you make this the only declaration available? In other words, give the original declaration a second template argument `spacedim`?